spec/models/government_spec.rb

```diff
@@ -1,104 +1,37 @@
-    end
-
-    it "raises an error if there isn't a government for the id" do
-      missing_content_id = SecureRandom.uuid
-      expect { Government.find(missing_content_id) }
-        .to raise_error(RuntimeError, "Government #{missing_content_id} not found")
-    end
-  end
-
-  describe ".for_date" do
-    let(:current_government) do
-      build(:government,
-            start_date: Date.parse("2018-12-01"),
-            end_date: nil)
-    end
-
-    let(:past_government) do
-      build(:government,
-            start_date: Date.parse("2018-01-31"),
-            end_date: Date.parse("2018-12-01"))
-    end
-
-    before do
-      allow(Government).to receive(:all)
-                       .and_return([current_government, past_government])
-    end
-
-    it "returns a government when there is one for the date" do
-      expect(Government.for_date(Date.parse("2019-06-01"))).to eq current_government
-    end
-
-    it "returns nil when there isn't a government for the date" do
-      expect(Government.for_date(Date.parse("2017-06-01"))).to be_nil
-    end
-
-    it "opts for the more recent government when there is a date conflict" do
-      expect(Government.for_date(Date.parse("2018-12-01"))).to eq current_government
-    end
-
-    it "accepts a nil date" do
-      expect(Government.for_date(nil)).to be_nil
-    end
-  end
-
-  describe ".current" do
-    let(:current_government) do
-      build(:government,
-            start_date: Date.parse("2018-12-01"),
-            end_date: nil)
-    end
-
-    let(:past_government) do
-      build(:government,
-            start_date: Date.parse("2018-01-31"),
-            end_date: Date.parse("2018-12-01"))
-    end
+  describe "#==" do
+    it "returns true when content_id and locale are equal" do
+      government = build(:government, started_on: Time.zone.today)
+      other = build(:government,
+                    content_id: government.content_id,
+                    locale: government.locale,
+                    started_on: Time.zone.yesterday)
 
-    before do
-      allow(Government).to receive(:all)
-                       .and_return([current_government, past_government])
+      expect(government).to eq(other)
     end
 
-    it "returns the current government" do
-      expect(Government.current).to eq current_government
-    end
-  end
+    it "returns false when content_id and locale there are not" do
+      content_id = SecureRandom.uuid
+      english = build(:government, content_id: content_id, locale: "en")
+      french = build(:government, content_id: content_id, locale: "fr")
+      expect(english).not_to eq(french)
 
-  describe ".past" do
-    it "returns the past governments" do
-      past = Government.past
-      expect(past).not_to be_empty
-      expect(past).to all be_a(Government)
-      expect(past).not_to include(Government.current)
+      government = build(:government, content_id: SecureRandom.uuid, locale: "en")
```

Comment:
Not sure what you're suggesting here?

Reply:
After fixing the test name to `it "returns false when content_id matches but locales don't" do`, you'll have something like this inside the block:

```ruby
  content_id = SecureRandom.uuid
  english = build(:government, content_id: content_id, locale: "en")
  french = build(:government, content_id: content_id, locale: "fr")
  expect(english).not_to eq(french)

  government = build(:government, content_id: SecureRandom.uuid, locale: "en")
  other = build(:government, content_id: SecureRandom.uuid, locale: "en")
  expect(government).not_to eq(other)
```

My suggestion is that these last three lines should be in their own test: "returns false when content_id doesn't match", and that that test should come _above_ the "returns false when content_id matches but locales don't"  test (since that one is more specialised). Hope that makes sense!

Reply:
Ah ok sure. Yeah I never have much fun on deciding what to test when you've got a simple and condition as you effectively have 4 different scenarios but feels overkill to test them all. Goes even worse with 3 items in a condition 😬